### DISCUSSION TITLE
Plugin code rewrite

### DISCUSSION THREAD

**Discussion by KygekDev** [Kygekraqmak/KygekJoinUI#13] [Ideas]
The source code is just too messy it needs a rewrite, so I'm going to rewrite it very soon.

I'm also planning to introduce the long-awaited commands feature alongside this update. I'm unsure whether to increment the major version or just the minor version for the rewrite, but it's unlikely it will be required to increment the major version, unless it contains breaking changes (which won't exist anyway because the plugin doesn't have any public APIs).

If you agree to this update, please vote with 👍, otherwise vote with 👎. Any questions and suggestions are welcome as long as it's useful to me and to the community.

**Comment by NhanAZ** [comment]
Only one vote!!!

**Reply by KygekDev** [reply]
Don't worry, the post is still relatively new. I'll give a chance around 1 month for everyone to vote and give suggestions, after that I'll lock this discussion
